spinnaker.md

```diff
@@ -0,0 +1,268 @@
+kind: ClusterRoleBinding
+metadata:
+ name: spinnaker-robot
+roleRef:
+ apiGroup: rbac.authorization.k8s.io
+ kind: ClusterRole
+ name: cluster-admin
+subjects:
+- kind: ServiceAccount
+  name: spinnaker-robot
+  namespace: default 
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+ name: spinnaker-robot
+ namespace: default
+EOF
+```
+
+Apply the manifest: 
+
+```
+kubectl apply -f spinnaker.yaml
+```
+
+### Extract  token from the secret associated with the Spinnaker service account: 
+
+List the secrets with:
+
+```
+kubectl get secrets
+```
+
+The output should include a secret named similarly to **spinnaker-robot-token-xxxx**.  Copy that token name for use below.
+
+Retrieve the token from the s**pinnaker-robot-token-xxxx **using the following command:
+
+```
+kubectl get secret <secret name> -o jsonpath="{['data']['token']}" | base64 -D
+```
+
+Update `kubeconfig`:
+
+
+> Create a backup of your kubeconfig, e.g. cp ~/.kube/config ~/.kube/config.old, before making the following modifications. 
+
+```
+users:
+- name: aws
+  user:
+    token: <token_string> 
+    exec:
+      apiVersion: client.authentication.k8s.io/v1alpha1
+      command: heptio-authenticator-aws
+      args:
+        - "token"
+        - "-i"
+        - "<eks_cluster_name>"
+```
+
+## Choose an environment
+
+This tells Halyard what type of environment to install Spinnaker into.  For this exercise we'll be using the distributed installation on Kubernetes method.  This method deploys each of Spinnaker's microservices separately.  
+
+Run the following command to add the environment: 
+
+```
+hal config deploy edit --type distributed --account-name eks-cluster
+```
+
+## Choose a storage service
+
+Spinnaker can be configured to use S3 to store all its persistent data in a S3 bucket. 
+
+To configure the S3 settings, run the following (notice, do not supply the value of `--secret-access-key` on the command line, you will be prompted to enter the value on STDIN once the command has started running):
+
+```
+hal config storage s3 edit \
+--access-key-id <aws_access_key_id> \
+--secret-access-key \
+--region <aws_region> 
+```
+
+> This assumes that you are logged in as a IAM user with administrator/root privileges
+
+
+Set the storage source to S3:
+
+```
+hal config storage edit --type s3
+```
+
+> If you get an S3 error such as “Access Denied”, insure that you have proper S3 bucket create and write access.
+
+## Deploy
+
+Now that we've finished setting the configuration parameters for Spinnaker, we're ready to deploy it and connect to it.  
+
+### Pick a version
+
+List the available versions:
+
+```
+hal version list
+```
+
+Set the version you want to use:
+
+```
+hal config version edit --version 1.8.0 
+```
+
+## Deploy Spinnaker
+
+```
+hal deploy apply
+```
+
+## Connect to the Spinnaker UI
+
+Spinnaker will not be publicly reachable by default so you will need to run the following command to reach the UI on [localhost:9000](http://localhost:9000/):
```

Comment:
Add a snapshot of the UI?

Reply:
There are lots of screenshots in the walk through.  I didn't really feel like it was necessary to add a screen shot here.